File: primero-server/src/main/java/inha/primero_server/domain/inquiry/dto/response/InquiryResponse.java
package inha.primero_server.domain.inquiry.dto.response;

import inha.primero_server.domain.inquiry.entity.enums.Status;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;
import java.util.List;

public record InquiryResponse (

    @Schema(description = "문의 ID", example = "1")
    Integer id,

    @Schema(description = "문의 제목", example = "교환 환불 문의")
    String title,

    @Schema(description = "신청 시간", example = "제품에 이상이 있어 환불 부탁드립니다.")
    String content,

    @Schema(description = "신청 시간", example = "OPEN")
    Status status,

    @Schema(description = "작성자", example = "Oh")
    String writer,

    @Schema(description = "답변 내용 리스트", example = "2024-08-01T00:00:00")
    List<AnswerResponse> answers,

    @Schema(description = "문의 등록 시간", example = "2024-08-01T00:00:00")
    LocalDateTime createdAt,

    @Schema(description = "문의 수정 시간", example = "2024-08-03T00:00:00")
    LocalDateTime lastModifiedAt
) {}

File: primero-client/lib/features/inquiry/models/inquiry_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'inquiry_response.freezed.dart';
part 'inquiry_response.g.dart';

@freezed
class InquiryResponse with _$InquiryResponse {
  const factory InquiryResponse({
    required int id,
    required String title,
    required String content,
    required String status,
    required String writer,
    required List<AnswerResponse> answers,
    required String createdAt,
    required String lastModifiedAt,
  }) = _InquiryResponse;

  factory InquiryResponse.fromJson(Map<String, dynamic> json) =>
      _$InquiryResponseFromJson(json);
}

@freezed
class AnswerResponse with _$AnswerResponse {
  const factory AnswerResponse({
    required int id,
    required String content,
    required String createdAt,
    required String lastModifiedAt,
  }) = _AnswerResponse;

  factory AnswerResponse.fromJson(Map<String, dynamic> json) =>
      _$AnswerResponseFromJson(json);
}
